broker/src/main/java/io/camunda/zeebe/broker/partitioning/PartitionManagerImpl.java

```diff
@@ -7,114 +7,214 @@
+    this.healthCheckService = healthCheckService;
+    this.deploymentRequestHandler = deploymentRequestHandler;
+    this.diskSpaceUsageListenerRegistry = diskSpaceUsageListenerRegistry;
+    this.commandHApiService = commandHApiService;
+    this.exporterRepository = exporterRepository;
+
+    partitionGroup =
+        new RaftPartitionGroupFactory().buildRaftPartitionGroup(brokerCfg, snapshotStoreFactory);
 
-    this.partitionGroup = Objects.requireNonNull(partitionGroup);
+    final var membershipService = clusterServices.getMembershipService();
+    final var communicationService = clusterServices.getCommunicationService();
 
-    partitions = buildPartitionService(membershipService, communicationService);
+    partitionService =
+        new DefaultPartitionService(membershipService, communicationService, partitionGroup);
 
     topologyManager = new TopologyManagerImpl(membershipService, localBroker);
+    partitionListeners.add(topologyManager);
+
+    this.partitionListeners = partitionListeners;
   }
 
   @Override
   public ManagedPartitionGroup getPartitionGroup() {
     return partitionGroup;
   }
 
-  public synchronized CompletableFuture<Void> start() {
+  public CompletableFuture<Void> start() {
     if (closeFuture != null) {
       return Futures.exceptionalFuture(
           new IllegalStateException(
-              "Atomix instance " + (closeFuture.isDone() ? "shutdown" : "shutting down")));
+              "PartitionManager " + (closeFuture.isDone() ? "shutdown" : "shutting down")));
     }
 
     actorSchedulingService.submitActor(topologyManager);
 
-    return partitions.start().thenApply(ps -> null);
+    return partitionService
+        .start()
+        .thenApply(
+            ps -> {
+              LOGGER.info("Registering Partition Manager");
+
+              /* this must be called here; it must be called after the members of the partition have been
+              populated which happens during start of bootstrap manager and before the partitions are registered */
+              healthCheckService.registerPartitionManager(this);
+
+              LOGGER.info("Starting partitions");
+
+              final var partitionFactory =
+                  new PartitionFactory(
+                      actorSchedulingService,
+                      brokerCfg,
+                      localBroker,
+                      deploymentRequestHandler,
+                      commandHApiService,
+                      snapshotStoreFactory,
+                      clusterServices,
+                      exporterRepository,
+                      healthCheckService);
+
+              partitions.addAll(
+                  partitionFactory.constructPartitions(
+                      partitionGroup, partitionListeners, this::addTopologyPartitionListener));
+
+              final var futures =
+                  partitions.stream()
+                      .map(partition -> CompletableFuture.runAsync(() -> startPartition(partition)))
```

Comment:
:wrench: IIRC, this will start the partition in the shared ForkJoin thread pool. We don't really use this typically, and I would avoid it if need be as it becomes difficult to know what is using it in a larger system. I get that this seems silly at bootstrap since not much should be using it, so we could always make an exception, but it should be commented or explained. That said, I'm fine merging this with the idea that we would move to using the actor scheduler for synchronization in a follow up issue, reusing our beautiful `Apfelkuchen` interface to await futures (since that's mostly what start partition does, right? Join on a future?). I'm happy to do that as a next issue, but if you want to go for it now I won't stop you :smile:

Reply:
Let's do it in the next issue :smile: